service.tf

```diff
@@ -15,17 +15,21 @@ resource "aws_ecs_service" "application" {
   health_check_grace_period_seconds  = var.healthcheck_grace
   scheduling_strategy                = var.scheduling_strategy
 
-  load_balancer {
-    container_name   = var.name
-    container_port   = var.port
-    target_group_arn = aws_lb_target_group.application.arn
+  dynamic "load_balancer" {
+    for_each = var.create_alb ? [1] : []
+
+    content {
+      container_name   = var.create_nginx ? var.nginx_container_name : var.name
+      container_port   = var.port
+      target_group_arn = aws_lb_target_group.application[0].arn
+    }
   }
 
   lifecycle {
     ignore_changes = [desired_count]
   }
 
-  placement_constraints = {
+  placement_constraints {
```

Comment:
ty placement constraints bych dal z tohoto modulu úplně pryč, my už to nepoužíváme a dělá to tam bordel, případně to dát zase jako optional.